src/rgw/rgw_op.cc

```diff
@@ -1390,16 +1445,36 @@ void RGWGetObj::execute()
 
   start = ofs;
 
+  op_ret = this->get_decrypt_filter(&decrypt, filter);
+  if (op_ret < 0) {
+    goto done_err;
+  }
+
+  if (need_decompress) {
+
+    if (cs_info.blocks.size() == 0) {
+      lderr(s->cct) << "ERROR: no info about compression blocks, cannot decompress" << dendl;
+      op_ret = -EIO;
+      goto done_err;
+    }
+
+    decompress = boost::in_place(s->cct, cs_info, partial_content, filter);
```

Comment:
I don't see `cs_info` being used after this point, consider passing it into the `RGWGetObj_Decompress` ctor by rvalue ref so we don't have to copy the `cs_info.blocks` vector?